packages/synapse-react-client/src/synapse-client/SynapseClient.ts

```diff
@@ -3056,6 +3059,34 @@ export const getAllOfPaginatedService = async <T>(
   return results
 }
 
+export type FunctionReturningNextPageToken<T> = (
+  nextPageToken?: string | null,
+) => Promise<{ results: T[]; nextPageToken?: string | null }>
+
+export async function getAllOfNextPageTokenPaginatedService<T>(
```

Comment:
See lines immediately above this for `getAllOfPaginatedService`, which pretty much works the same but with a different response body interface